src/MCPServer/lib/server/packages.py

```diff
@@ -758,6 +758,29 @@ class Transfer(Package):
     UNIT_VARIABLE_TYPE = "Transfer"
     JOB_UNIT_TYPE = "unitTransfer"
 
+    @classmethod
+    @auto_close_old_connections()
+    def get_or_create_from_db_by_path(cls, path):
+        """Matches a directory to a database Transfer by its appended UUID, or path."""
+        path = path.replace(settings.SHARED_DIRECTORY, r"%sharedPath%", 1)
+
+        transfer_uuid = uuid_from_path(path)
+        if transfer_uuid:
+            transfer_obj, created = models.Transfer.objects.get_or_create(
+                uuid=transfer_uuid, defaults={"currentlocation": path}
+            )
+            logger.info("Updating Transfer %s path: %s", transfer_obj.uuid, path)
+            if not created and transfer_obj.currentlocation != path:
+                transfer_obj.currentpath = path
+                transfer_obj.save()
+        else:
+            transfer_obj = models.Transfer.objects.create(
+                uuid=uuid4(), currentlocation=path
+            )
+            logger.info("Creating Transfer %s at %s", transfer_obj.uuid, path)
```

Comment:
Would't the log entry be inaccurate if uuid_from_path returns a truthy value and `created=True`? Maybe a single log call after the if-else, having `created = True` declared earlier would fix that?

```python
logger.info(
    "Creating %s %s at %s",
    transfer_obj.uuid, "created" if created else "updated", path)
```